core/src/main/scala/chisel3/BundleComparator.scala

```diff
@@ -0,0 +1,39 @@
+package chisel3
+
+/** This comparator compares the new and old elements accessor of a [[Bundle]]'s fields
+  * It lives here at the present to make working on the compiler plugin easier to
+  * test without having to recompile all tests in order to test the plugin.
+  * It would be nice to have tests in the plugin but that is hard to do
+  */
+object BundleComparator {
+
+  /**
+    * @param bundle         the bundle that is being tested
+    * @param showComparison shows the new and old fields side by side
+    * @return               returns true if a discrepancy is found
+    */
+  def apply(bundle: Bundle, showComparison: Boolean = false): Boolean = {
+    val header = s"=== Bundle Comparator ${bundle.className} " + "=" * 40
+    if (showComparison) {
+      println(header)
+      println(f"${"New Field Name"}%30s ${"id"}%6s ${"Old Field Name"}%30s ${"id"}%6s")
+    }
+
+    val newElements = bundle.elements.toList
+    val oldElements = bundle.oldElementsNoChecks.toList
+
+    var discrepancyFound = false
+    newElements.zipAll(oldElements, "Oops" -> Bool(), "Oops" -> Bool()).foreach {
+      case ((a, b), (c, d)) =>
+        val color = if (a == c) { Console.RESET }
+        else { discrepancyFound = true; Console.RED }
+        if (showComparison) {
+          println(f"$color$a%30s (${b._id}%06x) $c%30s (${d._id}%06x) ${Console.RESET}")
+        }
+    }
+    if (showComparison) {
+      println("=" * header.length)
+    }
+    discrepancyFound
+  }
+}
```

Comment:
This definitely needs to not be part of the public API, if it's absolutely needed it should be in package `chisel3.internal`. That being said, I don't think this is necessary to keep. If you are that worried about needing it for debugging future issues, it would be better if we can move this and the implementation of the old elements into the test project ([TestUtils](https://github.com/chipsalliance/chisel3/blob/master/src/test/scala/chisel3/testers/TestUtils.scala) is in package chisel3 so can access package private stuff like `getPublicFields`).

Reply:
I've removed this